pkg/providers/imagefamily/azlinux.go

```diff
@@ -68,12 +71,13 @@ func (u AzureLinux) DefaultImages() []DefaultImageOutput {
 				scheduling.NewRequirement(v1.LabelArchStable, v1.NodeSelectorOpIn, corev1beta1.ArchitectureArm64),
 				scheduling.NewRequirement(v1alpha2.LabelSKUHyperVGeneration, v1.NodeSelectorOpIn, v1alpha2.HyperVGenerationV2),
 			),
+			Distro: "aks-azurelinux-v2-arm64-gen2",
 		},
 	}
 }
 
 // UserData returns the default userdata script for the image Family
-func (u AzureLinux) UserData(kubeletConfig *corev1beta1.KubeletConfiguration, taints []v1.Taint, labels map[string]string, caBundle *string, _ *cloudprovider.InstanceType) bootstrap.Bootstrapper {
+func (u AzureLinux) SelfContainedUserData(kubeletConfig *corev1beta1.KubeletConfiguration, taints []v1.Taint, labels map[string]string, caBundle *string, _ *cloudprovider.InstanceType) bootstrap.Bootstrapper {
```

Comment:
If we are going to rename this and change from the AWS Paradigm we should call this CustomData instead of UserData? 

This ends up getting converted to custom data here https://github.com/Azure/karpenter-provider-azure/blob/main/pkg/providers/instance/instance.go#L336

The reason we called it user data was to align with AWS but if we are changing might as well go all the way.

Reply:
According to self-contained feature owners, right now that is actually `CustomData` (yes, I think we've been mislabeling it). But there could be a switch soon.